packages/block-library/src/block/edit.js

```diff
@@ -1,49 +1,80 @@
 /**
  * External dependencies
  */
-import { noop, partial } from 'lodash';
+import { partial } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { Component, Fragment } from '@wordpress/element';
 import { Placeholder, Spinner, Disabled } from '@wordpress/components';
-import { withSelect, withDispatch } from '@wordpress/data';
+import {
+	withSelect,
+	withDispatch,
+	RegistryProvider,
+	RegistryConsumer,
+	createRegistry,
+} from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { BlockEdit } from '@wordpress/block-editor';
-import { compose } from '@wordpress/compose';
+import { BlockEditorProvider, BlockList, storeConfig } from '@wordpress/block-editor';
+import { compose, createHigherOrderComponent } from '@wordpress/compose';
+import { parse, serialize } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import ReusableBlockEditPanel from './edit-panel';
 import ReusableBlockIndicator from './indicator';
 
+/**
+ * Higher-order component which renders the original component with the current
+ * registry context passed as its `registry` prop.
+ *
+ * @param {WPComponent} OriginalComponent Original component.
+ *
+ * @return {WPComponent} Enhanced component.
+ */
+const withRegistry = createHigherOrderComponent(
```

Comment:
This could be provided by default by the data module, it's used in at least two places.